var/spack/repos/builtin/packages/py-non-regression-test-tools/package.py

```diff
@@ -18,7 +18,8 @@ class PyNonRegressionTestTools(PythonPackage):
 
     version("develop", branch="develop")
     version("main", branch="main")
-    version("1.1.2", tag="v1.1.2", preferred=True)
+    version("1.1.2", tag="v1.1.2")
+    version("1.1.4", tag="v1.1.4", preferred=True)
```

Comment:
Newer versions should be listed above older versions. `preferred=True` is unneeded here since `main` and `develop` are never picked over numbered versions.

Reply:
Thanks for Looking this PR! Changes are made in  https://github.com/spack/spack/commit/4b5b1c2e6295ce20ebb993d1047773e51996d1d9